polling_stations/apps/data_finder/helpers/directions.py

```diff
@@ -97,12 +98,67 @@ def get_route(self, start, end):
         )
 
 
+class MapboxDirectionsClient(DirectionsClient):
+    precision = 5
+
+    def get_data(self, url):
+        resp = requests.get(url)
+        if resp.status_code != 200:
+            raise DirectionsException(
+                "Mapbox Directions API error: HTTP status code %i" % resp.status_code
+            )
+        return resp.json()
+
+    def get_route(self, start, end):
+        distance_km = get_distance(start, end)
+        if distance_km > 1.5:
+            transport_verb = {"base": "drive", "gerund": "driving-traffic"}
+        else:
+            transport_verb = {"base": "walk", "gerund": "walking"}
+
+        url = "{base_url}/{profile}/{coordinates}?&alternatives=false&geometries=polyline&steps=false&access_token={key}".format(
+            base_url=settings.BASE_MAPBOX_URL,
+            profile=transport_verb["gerund"],
+            coordinates="{start_lon},{start_lat};{end_lon},{end_lat}".format(
+                start_lon=start.x, start_lat=start.y, end_lon=end.x, end_lat=end.y
+            ),
+            key=settings.MAPBOX_API_KEY,
+        )
+
+        directions = self.get_data(url)
+
+        if directions["code"] != "Ok":
+            raise DirectionsException(
+                "Mapbox Directions API error: {}".format(directions["code"])
+            )
+
+        route = directions["routes"][0]["geometry"]
+
+        time = f'{math.ceil(directions["routes"][0]["duration"]/60)} {_("minute")}'
+
+        dist = f'{math.ceil(directions["routes"][0]["distance"]*0.0006213712)} {_("miles")}'
```

Comment:
0.0006213712 looks like a magic number; is it metres to miles?

Reply:
Yep - is there a better way of doing this?

Reply:
If you give it a name (a const or a function) then it will be clearer what `0.0006213712` is/means when someone comes to read this in future. For example:

```
def metres_to_miles(metres):
    return metres*0.0006213712
```

`dist = math.ceil(metres_to_miles(directions["routes"][0]["distance"]))`

documents this

Reply:
I've picked this up in https://github.com/DemocracyClub/UK-Polling-Stations/blob/translation-fixes-2021-04-29/polling_stations/apps/data_finder/helpers/directions.py#L22-L24.

Both Google and Mapbox return a numerical distance in metres, so we can put that in our Direction object and fetch the more usual unit from a property in the template.